File: fsharp/fac.fs
// Learn more about F# at http://fsharp.org

open System

let fac n =
        seq { 1..n }
        |> Seq.map bigint
        |> Seq.reduce(*)

[<EntryPoint>]
let main argv =
    let res = seq { 1..3000 } |> Seq.map fac |> Seq.sum
    printf "%A" res
    0 // return an integer exit code

File: erlang/fac.erl
#!/usr/bin/env escript

main([]) ->
    S = lists:sum([fac(X) || X <- lists:seq(1,3000)]),
    io:write(S).

fac(N) ->
    lists:foldl(fun(X, Fac) -> X * Fac end, 1, lists:seq(1,N)).
